### PR TITLE
Update to monaco-editor@0.43.0 and monaco-vscode-api@1.82.0

### PR DESCRIPTION
**Pull request by kaisalmen**
Changes:
- Update all dependencies:
  - `monaco-editor@0.43.0`
  - `monaco-vscode-api@1.82.0`
  - Multiple others including `typescript@5.2.2`
- Removed dynamic imports:
   - `initServices` add both language and model service always and they are therefore a direct dependency of `monaco-languageclient`. Without them using this lib doesn't really make sense.
   - The examples package define the extra services and default-extensions the examples require in their dependencies

TODO:
- [x] Update README
- [x] Use final version of `@codingame/monaco-vscode-api` as soon as it released.

### CODE REVIEW

**Comment by kaisalmen** [comment]
@CGNonofr it is working nicely after I adapted the vite / experimental node config

**Comment by kaisalmen** [comment]
Next version is available: https://www.npmjs.com/package/monaco-languageclient/v/6.5.0-next.0

**Comment by CGNonofr** [comment]
> @CGNonofr it is working nicely after I adapted the vite / experimental node config

Wonderful! Don't you want me to wait for the monaco-vscode-api final release before approving it?

**Comment by kaisalmen** [comment]
> Wonderful! Don't you want me to wait for the monaco-vscode-api final release before approving it?

Yes, I also want/force-me to update the README on this branch. 🙂

**Comment by kaisalmen** [comment]
@CGNonofr updated to `1.82.2`
Next version is available: https://www.npmjs.com/package/monaco-languageclient/v/6.5.0-next.1

The README is basically ready. Should we remove the [monaco-editor-core](https://github.com/TypeFox/monaco-languageclient#monaco-editor-core) section from the troubleshooting section. This no longer relevant. WDYT?

**Comment by kaisalmen** [comment]
@CGNonofr this is now ready to be merged. Further README clarification don't block this, I think.

File: packages/client/package.json [hunk 1]
```diff
@@ -48,14 +48,14 @@
...
+    "monaco-editor": "~0.43.0",
+    "vscode": "npm:@codingame/monaco-vscode-api@1.82.0-next.0",
     "vscode-jsonrpc": "~8.1.0",
     "vscode-languageclient": "~8.1.0"
   },
   "peerDependencies": {
-    "monaco-editor": "~0.41.0",
-    "vscode": "npm:@codingame/monaco-vscode-api@1.81.7"
+    "monaco-editor": "~0.43.0",
+    "vscode": "npm:@codingame/monaco-vscode-api@1.82.0-next.0"
```
Comment (CGNonofr) [inline packages/client/package.json]: forgot to switch on the final version?

File: README.md [hunk 2]
```diff
@@ -282,7 +283,7 @@ loader.config({ monaco });
 If you use pnpm, you have to add `vscode` / `monaco-vscode-api` as direct dependency (see the [following table](#monaco-editor--monaco-vscode-api-compatibility-table)), otherwise the installation will fail.
 
 ```json
-"vscode": "npm:@codingame/monaco-vscode-api@1.81.7"
+"vscode": "npm:@codingame/monaco-vscode-api@1.82.0"
```
Comment (CGNonofr) [inline README.md]: 1.82.0 failed :x the first valid version is 1.82.1, no change with 1.82.2

File: packages/client/package.json (hunk 1 above)
Comment (kaisalmen) [reply packages/client/package.json]: You must have checked an outdated version. This is `1.82.2`

File: README.md (hunk 2 above)
Comment (kaisalmen) [reply README.md]: You must have checked an outdated version. This is `1.82.2`

Comment (CGNonofr) [reply README.md]: indeed, no idea what happened :shrug:

**Review by CGNonofr — APPROVED** [review]

**Comment by kaisalmen** [comment]
@CGNonofr I have not released a final version, because I discovered a nasty problem in `monaco-editor-wrapper` and here is the solution:
https://github.com/TypeFox/monaco-components/blob/next-mlc/packages/examples/src/langium/config/wrapperLangiumVscode.ts#L70
https://github.com/TypeFox/monaco-components/blob/next-mlc/packages/monaco-editor-wrapper/src/editorAppVscodeApi.ts#L69-L94
When you load an extension for a language (Langium Web Worker LS) you now have to await the loading of the theme before you load the language extension. Otherwise the theme does not work. This behaviour changed from with `monaco-vscode-api@1.82.x`. It was no problem before. Are you aware of this change in behaviour?

**Comment by kaisalmen** [comment]
@CGNonofr the problem is that sometime the loading of extensions (e.g. themes) happens before and sometimes after. This seems that something that has been awaited before (1.81.x) and now (1.82.x). Maybe something needs to be adjusted there. It is anyway good I have the other mechanism in place above.

**Pull request merged**